test/unit/org/apache/cassandra/db/ScrubTest.java

```diff
@@ -838,42 +853,117 @@ public void testNoCheckScrubMultiRowWithTool()
         Assertions.assertThat(tool.getStdout()).contains("1 rows in new sstable and 0 empty");
         tool.assertOnCleanExit();
         assertOrderedAll(cfs, 1);
+    }
+
+    @Test
+    public void testHeaderFixFixOnlyWithTool()
+    {
+        String ksName = Thread.currentThread().getStackTrace()[1].getMethodName();
+        toolTestingSetup(ksName);
 
-        tool = ToolRunner.invokeClass(StandaloneScrubber.class, "-e", "fix-only", KEYSPACE2, CF3);
+        CompactionManager.instance.disableAutoCompaction();
+        Keyspace keyspace = Keyspace.open(ksName);
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF3);
+        cfs.clearUnsafe();
+
+        fillCF(cfs, 1);
+        assertOrderedAll(cfs, 1);
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneScrubber.class, "-e", "fix-only", ksName, CF3);
         Assertions.assertThat(tool.getStdout()).contains("Not continuing with scrub, since '--header-fix fix-only' was specified.");
         tool.assertOnCleanExit();
         assertOrderedAll(cfs, 1);
+    }
+
+    @Test
+    public void testHeaderFixWithTool()
+    {
+        String ksName = Thread.currentThread().getStackTrace()[1].getMethodName();
+        toolTestingSetup(ksName);
+
+        CompactionManager.instance.disableAutoCompaction();
+        Keyspace keyspace = Keyspace.open(ksName);
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF3);
+        cfs.clearUnsafe();
+
+        fillCF(cfs, 1);
+        assertOrderedAll(cfs, 1);
 
-        tool = ToolRunner.invokeClass(StandaloneScrubber.class, "-e", "fix", KEYSPACE2, CF3);
+        ToolResult tool = ToolRunner.invokeClass(StandaloneScrubber.class, "-e", "fix", ksName, CF3);
         Assertions.assertThat(tool.getStdout()).contains("Pre-scrub sstables snapshotted into");
         Assertions.assertThat(tool.getStdout()).contains("1 rows in new sstable and 0 empty");
         tool.assertOnCleanExit();
         assertOrderedAll(cfs, 1);
+    }
+
+    @Test
+    public void testHeaderFixNoChecksWithTool()
+    {
+        String ksName = Thread.currentThread().getStackTrace()[1].getMethodName();
+        toolTestingSetup(ksName);
 
-        tool = ToolRunner.invokeClass(StandaloneScrubber.class, "-e", "off", KEYSPACE2, CF3);
+        CompactionManager.instance.disableAutoCompaction();
+        Keyspace keyspace = Keyspace.open(ksName);
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF3);
+        cfs.clearUnsafe();
+
+        fillCF(cfs, 1);
+        assertOrderedAll(cfs, 1);
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneScrubber.class, "-e", "off", ksName, CF3);
         Assertions.assertThat(tool.getStdout()).contains("Pre-scrub sstables snapshotted into");
         Assertions.assertThat(tool.getStdout()).contains("1 rows in new sstable and 0 empty");
         tool.assertOnCleanExit();
         assertOrderedAll(cfs, 1);
     }
+
+    private void toolTestingSetup(String ksName)
```

Comment:
We could create the keyspace name here, and possibly also disable autocompaction, so we reduce duplication in the callers:
```java
private String toolTestingSetup()
{
    System.setProperty(org.apache.cassandra.tools.Util.ALLOW_TOOL_REINIT_FOR_TEST, "true"); // Necessary for testing
    CompactionManager.instance.disableAutoCompaction();

    String ksName = Thread.currentThread().getStackTrace()[2].getMethodName();
    createKeyspace(ksName,
                   KeyspaceParams.simple(1),
                   standardCFMD(ksName, CF),
                   standardCFMD(ksName, CF2),
                   standardCFMD(ksName, CF3),
                   standardCFMD(ksName, CF4),
                   counterCFMD(ksName, COUNTER_CF));

    return ksName;
}
```